be/src/runtime/fragment_mgr.cpp

```diff
@@ -330,7 +330,9 @@ std::string FragmentMgr::to_http_path(const std::string& file_name) {
 void FragmentMgr::coordinator_callback(const ReportStatusRequest& req) {
     DCHECK(req.status.ok() || req.done); // if !status.ok() => done
     Status exec_status = req.update_fn(req.status);
-
+    if (_exec_env->new_load_stream_mgr()->have_promise(req.query_id)) {
```

Comment:
do not depend on this to set promise. coordinator_callback only means the fragment is finished on be, but there are some state on fe. 
Add a rpc method in be, and if fe find the coord finished, then fe call this rpc service to indicate the load finished.